hatbot.gs

```diff
@@ -0,0 +1,175 @@
+  var service = getService();
+  var authorized = service.handleCallback(request);
+  if (authorized) {
+    return HtmlService.createHtmlOutput('Success!');
+  } else {
+    return HtmlService.createHtmlOutput('Denied');
+  }
+}
+
+function get_weather_data(location) {
+  
+  var response = UrlFetchApp.fetch("api.openweathermap.org/data/2.5/weather?q="+ location +"&appid=26d507c1547e5aba63c27c987cd19aa2");
+  var json = response.getContentText();
+  var data = JSON.parse(json);
+
+  Logger.log(data);
+
+  return data;
+}
+
+function hat_or_no_hat() {
+  var cur_weather = get_weather_data(where_am_i);
+  var decision = 0;
+  //strip data
+  var tempK = cur_weather["main"]["temp"];
+  var wind = cur_weather["wind"]["speed"];
+  
+  //fix units and clean data
+  tempK = tempK.toFixed(2);
+  var tempC = (tempK - 273.15).toFixed(2);
+
+  windKM = (wind * 3.6).toFixed(0)
+
+  //make a decision using if's and comparisons
+
+  if (tempC > too_hot_for_hat) {
+    hat_hot = true;
+  }
+  else {
+    hat_hot = false;
+  }
+  if (windKM > too_windy_for_hat) {
+    hat_windy = true;
+  }
+  else {
+    hat_windy = false;
+
+  }
+  // hot and windy = 1
+  // hot and calm = 2
+  // cool and windy = 3
+  // cooooooool and caaaaaaaaalm = 4
+  if ( hat_hot == true && hat_windy == true ){decision = 1; }
+  if ( hat_hot == true && hat_windy == false ){decision = 2; }
+  if ( hat_hot == false && hat_windy == true ){decision = 3; }
+  if ( hat_hot == false && hat_windy == false ){decision = 4; }
+  
+  Logger.log(tempC);
+  Logger.log(windKM);
+
+  Logger.log(hat_hot);
+  Logger.log(hat_windy);
+  
+  Logger.log(decision);
+
+  return decision;
+}
+
+function write_tweet(){
+  //take the decision from hat_or_no_hat and do something with it
+  var what_do_i_tweet = hat_or_no_hat();
+  var str = "";
+  var rand = 0;
+
+  if (what_do_i_tweet == 1){
+    rand = Math.floor(Math.random()*hot_and_windy.length);
+    str = hot_and_windy[rand];
+  }
+  if (what_do_i_tweet == 2){
+    rand = Math.floor(Math.random()*hot_and_calm.length);
+    str = hot_and_calm[rand];
+  }
+  if (what_do_i_tweet == 3){
+    rand = rand = Math.floor(Math.random()*cold_and_windy.length);
+    str = cold_and_windy[rand];
+  }
+  if (what_do_i_tweet == 4){
+    rand = Math.floor(Math.random()*ideal_hat_scenario.length);
+    str = ideal_hat_scenario[rand];
+  }
+  
+  return str;
+
+};
+
+function run() {
+  var service = getService();
+  var tweet = bot_name + write_tweet();
+  Logger.log(tweet);
+  if (service.hasAccess()) {
+    var url = 'https://api.twitter.com/1.1/statuses/update.json';
+    var payload = {
+      status: tweet
+    };
+    var response = service.fetch(url, {
+      method: 'post',
+      payload: payload
+    });
+    var result = JSON.parse(response.getContentText());
+    Logger.log(JSON.stringify(result, null, 2));
+  } else {
+    var authorizationUrl = service.authorize();
+    Logger.log('Open the following URL and re-run the script: %s',
```

Comment:
This would be Logger.error, this will make it clear something has gone wrong and needs attention.